pkg/apis/options/legacy_options.go

```diff
@@ -508,6 +508,7 @@ type LegacyProvider struct {
 	ApprovalPrompt                     string   `flag:"approval-prompt" cfg:"approval_prompt"` // Deprecated by OIDC 1.0
 	UserIDClaim                        string   `flag:"user-id-claim" cfg:"user_id_claim"`
 	AllowedGroups                      []string `flag:"allowed-group" cfg:"allowed_groups"`
+	AllowedRoles                       []string `flag:"allowed-role" cfg:"allowed_roles"`
```

Comment:
> Perhaps we want to keep it generic so that is can be used by multiple providers in the future 🤔

That was my intent - for commonly named things, keep the naming generic so it is easy to configure & document if multiple providers could potentially implement (this moves away from some of the conventions past providers have used, though).

I think this makes the transition to structured configs when this setting would be a child of a provider settings block a little clearer -- can keep the same name and not need to remove a provider specific prefix.

Reply:
I agree, `allowed-roles` does not have to be keycloak specific, other providers could also make use of it. Is that ok for you @JoelSpeed or would you prefer it to be prefixed with keycloak?

Reply:
My only concern here is that we should document which providers it works for, otherwise we will be in a situation as with many of our issues, where people try the feature only to realise it doesn't work with their provider. Let's make an issue to make sure the provider features are tallied up and documented per provider

Reply:
Hi, I understand your concerns and it might pose a risk of unintended behavior.
Still I think that especially for Keycloak there is a difference between group membership and roles (realm and client level). 

In our usecase we use roles for permissions -> for this proxy this means „you can pass or you shall not pass“ and we use groups as a manner of managing users (all members of a group have the same permissions but multiple groups can have those permissions). 

I have seen other users and implementations that use the „groups“ for permissions. 

On top of that at least Keycloak will allow us to design the tokens the way we want to… so there is a lot that can be done wrong.

The implementation of @pb82 leverages the default behavior of Keycloak and where it would map groups and roles into the token. So I think for Keycloak it is alright to use „role:“ as a delimiter. But: Keycloak does allow me to create a group „role:admin“ and I am not aware of any way around this. 
One Idea would be to configure if we want to authorize based upon groups or on roles.

One extr thing, that I could imagine is that, if we went for a completely independent OIDC JWT implementation we could configure a JSON Path to the claim in the token, we want to verify and then configure what values would be required to let the user pass.